src/backend/upload_file/upload_file_for_chunks.py

```diff
@@ -0,0 +1,207 @@
+        elif filename.endswith(".txt"):
+            approaches.append("Approach 3: File is a TXT based on extension")
+            file_type = "txt"
+        else:
+            approaches.append("No match: Unsupported file extension")
+
+        # Approach 2: Check content by file signature (simplified)
+        if filename.endswith(".pdf"):
+            try:
+                with open(filename, "rb") as f:
+                    header = f.read(4)
+                    if header == b"%PDF":
+                        approaches.append("Approach 2: PDF signature matched")
+            except Exception:
+                approaches.append("Approach 2: PDF signature not matched")
+
+        elif filename.endswith(".docx"):
+            try:
+                with open(filename, "rb") as f:
+                    header = f.read(4)
+                    if header == b"PK\x03\x04":
+                        approaches.append("Approach 2: DOCX signature matched")
+            except Exception:
+                approaches.append("Approach 2: DOCX signature not matched")
+
+        elif filename.endswith(".txt"):
+            try:
+                with open(filename, "r") as f:
+                    content = f.read(100)
+                    if content.isprintable():
+                        approaches.append("Approach 2: TXT content matched")
+            except Exception:
+                approaches.append("Approach 2: TXT content not matched")
+
+        # Approach 3: Check using libraries
+        if filename.endswith(".pdf"):
+            try:
+                with open(filename, "rb") as f:
+                    reader = PyPDF2.PdfReader(f)
+                    if len(reader.pages) > 0:
+                        approaches.append("Approach 3: PDF content extracted")
+            except Exception:
+                approaches.append("Approach 3: PDF content extraction failed")
+
+        elif filename.endswith(".docx"):
+            try:
+                doc = Document(filename)
+                if len(doc.paragraphs) > 0:
+                    approaches.append("Approach 3: DOCX content extracted")
+            except Exception:
+                approaches.append("Approach 3: DOCX content extraction failed")
+
+        elif filename.endswith(".txt"):
+            try:
+                with open(filename, "r") as f:
+                    content = f.read(100)
+                    if content:
+                        approaches.append("Approach 3: TXT content validated")
+            except Exception:
+                approaches.append("Approach 3: TXT content validation failed")
+
+        #Select valid approaches
+        valid_approaches = [approach for approach in approaches if "matched" in approach or "extracted" in approach]
+        if len(valid_approaches) >= 2:
+            return file_type
+        else:
+            raise HTTPException(status_code=400, detail="File type mismatch, extraction failed.")
+
+    def extract_pdf_content(self, file_data):
```

Comment:
For PDf content, make sure the below code is working and use the below code


import pdfplumber
import pytesseract
from PIL import Image
import json

def extract_pdf_with_layout(pdf_path, output_json_path):
    results = []

    with pdfplumber.open(pdf_path) as pdf:
        for page_number, page in enumerate(pdf.pages, start=1):
            page_elements = []

            # Extract text blocks with their bounding boxes
            for block in page.extract_words():
                page_elements.append({
                    "type": "text",
                    "content": block["text"],
                    "x0": block["x0"],
                    "y0": block["top"]
                })

            # Extract tables with their bounding boxes
            for table in page.extract_tables():
                table_text = format_table_as_text(table)
                bbox = page.find_tables()[0].bbox  # Get bounding box for the first table
                page_elements.append({
                    "type": "table",
                    "content": table_text,
                    "x0": bbox[0],
                    "y0": bbox[1]
                })

            # OCR for images or non-text content
            if not page.extract_text():
                page_image = page.to_image(resolution=300).original
                ocr_text = pytesseract.image_to_string(page_image)
                page_elements.append({
                    "type": "image",
                    "content": f"Data obtained from image: {ocr_text.strip()}",
                    "x0": 0,  # Assuming full-page images
                    "y0": 0
                })

            # Sort elements by their Y-coordinate and then X-coordinate for layout
            page_elements.sort(key=lambda e: (e["y0"], e["x0"]))

            # Combine elements into a single text flow
            page_text = " ".join([element["content"] for element in page_elements])

            # Append page result
            results.append({
                "page_number": page_number,
                "text": page_text
            })

    # Save to JSON
    with open(output_json_path, "w", encoding="utf-8") as output_file:
        json.dump(results, output_file, indent=4, ensure_ascii=False)

    print(f"Extraction complete. Data saved to {output_json_path}")


def format_table_as_text(table):
    """
    Convert a 2D table array into a neatly formatted text table.
    """
    col_widths = [max(len(str(item)) for item in col) for col in zip(*table)]
    row_format = " | ".join(f"{{:<{w}}}" for w in col_widths)
    formatted_table = "\n".join(row_format.format(*row) for row in table)
    return formatted_table.strip()


# Usage
pdf_file = "example.pdf"  # Input PDF
output_file = "output.json"  # Output JSON
extract_pdf_with_layout(pdf_file, output_file)